### ISSUE TITLE
Phase 0 Video Shows Incorrect ChessBoard Implementation

### ISSUE THREAD

**Issue by frozenfrank** [softwareconstruction240/softwareconstruction#139]
## Overview
As discussed in the the TA meeting on Sep 19, 2024, a significant number of students are being led astray by an incorrect implementation of a method in an instruction video. 

See the video at the [11:17 mark](https://byu.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=be427737-b0b5-4aec-bb15-b177014da69c) (linked to from the [Videos Section](https://github.com/softwareconstruction240/softwareconstruction/blob/main/chess/0-chess-moves/chess-moves.md#videos-2122) of the instruction): 
> All I have to do is come down here and actually implement putting that piece in the square... 
> Now I've successfully implemented adding that piece to it.

## Discussion
The video was only intended to give them a starting place and to still require them to think about their own issues, but it does go so far as to implement the method. It does serve as an instructional experience to cause students to think critically about the code they are writing; simply copying&pasting code from any source (ChatGTP, StackOverflow, students, professors, etc...) should never be acceptable.

## Potential Resolutions
1. Leave the error without changes
2. Leave the error but provide a warning that some of the code is incorrect
    * Explicitly tells students to be critically thinking about the code 
    * Turns it into a challenge to find the place that's wrong
    * Could be a written warning on the page that links to the video, or 
    * If the video is editable, it could be a warning inserted at the beginning of the video
3. Remake the video (or video segment) to simply _not implement_ the method
    * Requires students to think through the details from scratch
    * Avoids the potential misdirection of showing them code that is incorrect

**Comment by frozenfrank** [comment]
In the most recent TA meeting, we decided that #234 would be an appropriate resolution to this issue.

**Comment by TheDavSmasher** [comment]
Is this still relevant to do? What changes should be made?

**Comment by leesjensen** [comment]
I'm happy to rerecord the video. I can do this sometime in the next couple weeks. However, for the sake of consistency perhaps we want @rodham to create a video that matches all the other course videos?

**Comment by rodham** [comment]
We will eventually re-record this video when there are more significant changes needed, but I don't think this one issue is enough to justify that now. I think we should just add some text next to the video link explaining the issue.

**Comment by leesjensen** [comment]
@rodham, I rewatched the video and it is very out of date.

1. Old procedure for creating the repository (zip vs using template repo). This is the first half of the video.
2. Tests are in a different location. This was also wrong in the instruction. Great example of the maintenance costs of changes something small like the test directory structure.
3. Misleading index reference on board add method instruction.
4. Doesn't demonstrate creating toString/hashCode/equals.

I'm afraid the video is more confusing than it is helpful and it won't take me long to rerecord it. Are you OK with me doing that?

**Comment by rodham** [comment]
Yes, that would be great. When you're done, we can ask BYU Online to import
the new video into their video system and replace the old one. Thanks for
doing that.
